src/main/java/net/fabricmc/loader/discovery/ModResolver.java

```diff
@@ -85,6 +86,22 @@ private static IVecInt toVecInt(IntStream stream) {
 		return new VecInt(stream.toArray());
 	}
 
+	public static String getReadablePath(FabricLoader loader, ModCandidate c) {
+		Path path;
+		try {
+			path = UrlUtil.asPath(c.getOriginUrl());
+		} catch (UrlConversionException e) {
+			throw new RuntimeException(e);
+		}
+
+		Path gameDir = loader.getGameDirectory().toPath();
```

Comment:
```suggestion
		path = path.normalize();
		Path gameDir = loader.getGameDirectory().toPath().normalize();
```

On my system the game directory includes `/.` at the end (for some reason) - which should probably be handled here, or in `FabricLoader.getGameDirectory()`? (Either way normalizing the `path` is probably a good idea).

Reply:
I'd do it in `FabricLoader.setGameDir` or something, not special case it here.